pkg/apis/v1/ec2nodeclass.go

```diff
@@ -56,6 +56,10 @@ type EC2NodeClassSpec struct {
 	// +kubebuilder:validation:MaxItems:=30
 	// +optional
 	AMISelectorTerms []AMISelectorTerm `json:"amiSelectorTerms" hash:"ignore"`
+	// AMIFamily is the AMI family that instances use.
+	// +kubebuilder:validation:Enum:={AL2,AL2023,Bottlerocket,Custom,Windows2019,Windows2022}
+	// +optional
+	AMIFamily *string `json:"amiFamily,omitempty" hash:"ignore"`
```

Comment:
We could, I only hadn't since we hadn't previously. I can make the update.

Reply:
Honestly, this becomes rather cumbersome due to the type conversions, given we're converting to and from strings anyway, and the type alias doesn't really give us any additional time safety, I'd lean toward leaving it as is. I don't feel too strongly either way though.

Reply:
>I don't feel too strongly either way though

I think that we should add this as an enum long-term, but I'm fine not changing this logic right now and taking that as a tech debt item since it's more of a cleanup task